cloudinit/safeyaml.py

```diff
@@ -15,34 +15,101 @@
 # SchemaPathMarks track the path to an element within a loaded YAML file.
 # The start_mark and end_mark contain the row and column indicators
 # which represent the coordinates where the schema element begins and ends.
-SchemaPathMarks = namedtuple(
-    "SchemaPathMarks", ("path", "start_mark", "end_mark")
-)
+class SchemaPathMarks:
+    def __init__(self, path: str, start_mark: yaml.Mark, end_mark: yaml.Mark):
+        self.path = path
+        self.start_mark = start_mark
+        self.end_mark = end_mark
+
+    def __contains__(self, other):
+        """Return whether other start/end marks are within self marks."""
+        if (
+            other.start_mark.line < self.start_mark.line
+            or other.end_mark.line > self.end_mark.line
+        ):
+            return False
+        if (
+            other.start_mark.line == self.start_mark.line
+            and other.start_mark.column < self.start_mark.column
+        ):
+            return False
+        if (
+            other.end_mark.line == self.end_mark.line
+            and other.end_mark.column > self.end_mark.column
+        ):
+            return False
+        return True
+
+    def __eq__(self, other):
+        return (
+            self.start_mark.line == other.start_mark.line
+            and self.start_mark.column == other.start_mark.column
+            and self.end_mark.line == other.end_mark.line
+            and self.start_mark.line == other.start_mark.line
```

Comment:
These last 3 lines are all duplicate conditions, correct?

Reply:
ahh you are right, I've dropped two and changed the last one to `self.end_mark.column == other.end_mark.column`